### PR TITLE
fix: _report_checkpoint correct POST payload

### PR DESCRIPTION
**Pull request by mpkouznetsov**
## Description

Handle multiple issues:
1. Align the shape of `experimental.Checkpoint` with the Checkpoint protobuf representation.
2. Handle downstream issues detected by integration and e2e tests:
    a. As the generic json encoder does not handle some corner cases we could not rely on bindings to invoke Report Checkpoint endpoint. So we have to switch back to raw requests posting.
    b. Adjust an e2e test to reflect the fact that we now have `format`, `framework`, and `latest_batch` automatically added into the "free form" metadata dict.

## Commentary

Note: `XXX` prefix in comments is @rb-determined-ai 's way to designate unfinished CoreAPI items in `generic-checkpoints`.

### Checkpoint changes
| Old Checkpoint Attribute | Disposition |
| ------------------------|--------------|
| Checkpoint.uuid                |   |
| Checkpoint.experiment_config   | goes to training substruct |
| Checkpoint.experiment_id       | goes to training substruct |
| Checkpoint.trial_id            | goes to training substruct |
| Checkpoint.hparams             | goes to training substruct |
| Checkpoint.batch_number        | goes to metadata["latest_batch"] |
| Checkpoint.start_time          | deleted (not recorded anymore) |
| Checkpoint.end_time            | renamed to report_time |
| Checkpoint.resources           |   |
| Checkpoint.validation          | goes to training substruct |
| Checkpoint.framework           | goes to metadata["framework"] |
| Checkpoint.format              | goes to metadata["format"] |
| Checkpoint.determined_version  | goes to metadata["determined_version"] |
| Checkpoint.metadata            |  |
| Checkpoint.model_version       | deleted |
| Checkpoint.model_id            | deleted |

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/common/experimental/checkpoint/_checkpoint.py [hunk 1]
```diff
@@ -163,7 +164,7 @@ def download(self, path: Optional[str] = None) -> str:
         # export api, we need to start writing a different format (the user-defined metadata only),
         # and we need to start writing it at upload time, not download time.
         #
-        # see https://determined-ai.slack.com/archives/CSLAGUF3M/p1642188450030600?thread_ts=1642115346.016000&cid=CSLAGUF3M
+        # see https://rb.gy/vaxl7a
```
Comment (coreystatendet) [inline harness/determined/common/experimental/checkpoint/_checkpoint.py:167]: IMO we should avoid having links to private slack in the repo.  I know it's already landed, but can we copy or paraphrase the context here instead?

**Review by coreystatendet — COMMENTED** [review]
Looks good to me.  It looks like we're dropping model_id, framework, format, determined_version, and model_version from the checkpoint -- I don't have enough context to know why we would/wouldn't have those.  Are they superseded by e.g. metadata associated with taskId/allocationId?

**Comment by mpkouznetsov** [comment]
> Looks good to me. It looks like we're dropping model_id, framework, format, determined_version, and model_version from the checkpoint -- I don't have enough context to know why we would/wouldn't have those. Are they superseded by e.g. metadata associated with taskId/allocationId?

framework, format, determined_version are now part of the `metadata` dict. model_id and model_version are now excluded from the checkpoint (not all checkpoints have a corresponding registered model).
